keyboards/zsa/voyager/voyager.c

```diff
@@ -0,0 +1,301 @@
+// Copyright 2023 ZSA Technology Labs, Inc <@zsa>
+// Copyright 2023 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "voyager.h"
+
+keyboard_config_t keyboard_config;
+
+bool mcp23018_leds[2] = {0, 0};
+bool is_launching     = false;
+
+#ifdef DYNAMIC_MACRO_ENABLE
+static bool is_dynamic_recording = false;
+
+void dynamic_macro_record_start_user(void) {
+    is_dynamic_recording = true;
+}
+
+void dynamic_macro_record_end_user(int8_t direction) {
+    is_dynamic_recording = false;
+    STATUS_LED_3(false);
+}
+#endif
+
+void voyager_led_task(void) {
+    if (is_launching) {
```

Comment:
```suggestion
void voyager_led_task(void) {
    wait_ms(1); // ensure we don't hard-loop when neither launching nor recording.
    if (is_launching) {
```

Reply:
Alternatively stick it at the bottom of the function. Just need something to switch context when you're not launching or recording -- you're effectively hard-looping in the while loop in `LEDThread` when nothing is active.